problem_builder/completion.py

```diff
@@ -0,0 +1,137 @@
+# option) any later version of the AGPL published by the FSF.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Imports ###########################################################
+
+import logging
+import uuid
+
+from xblock.core import XBlock
+from xblock.fields import Scope, String, Boolean
+from xblock.fragment import Fragment
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+from xblockutils.resources import ResourceLoader
+
+from .mixins import QuestionMixin, XBlockWithTranslationServiceMixin
+from .sub_api import sub_api, SubmittingXBlockMixin
+
+
+# Globals ###########################################################
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+# Make '_' a no-op so we can scrape strings
+def _(text):
+    return text
+
+
+# Classes ###########################################################
+
+@XBlock.needs('i18n')
+class CompletionBlock(
+        SubmittingXBlockMixin, QuestionMixin, StudioEditableXBlockMixin, XBlockWithTranslationServiceMixin, XBlock
+):
+    """
+    An XBlock used by students to indicate that they completed a given task.
+    The student's answer is always considered "correct".
+    """
+    CATEGORY = 'pb-completion'
+    STUDIO_LABEL = _(u'Completion')
+    answerable = True
+
+    question = String(
+        display_name=_('Question'),
+        help=_('Mentions a specific activity and asks the student whether they completed it.'),
+        scope=Scope.content,
+        default=_(
+            'Please indicate whether you attended the In Person Workshop session by (un-)checking the option below.'
+        ),
+    )
+
+    answer = String(
+        display_name=_('Answer'),
+        help=_(
+            'Represents the answer that the student can (un-)check '
+            'to indicate whether they completed the activity that the question mentions.'
+        ),
+        scope=Scope.content,
+        default=_('Yes, I attended the session.'),
+    )
+
+    student_value = Boolean(
+        help=_("Records student's answer."),
+        scope=Scope.user_state,
+        default=None,
+    )
+
+    editable_fields = ('display_name', 'show_title', 'question', 'answer')
+
+    def mentoring_view(self, context):
+        """
+        Main view of this block.
+        """
+        context = context.copy() if context else {}
+        context['question'] = self.question
+        context['answer'] = self.answer
+        context['completion_id'] = 'pb-completion-{}'.format(uuid.uuid4().hex[:20])
```

Comment:
nit: The `completion_id` value is used for the HTML `id` attribute of the checkbox, but it doesn't look like we need the `id` attribute at all? I think we could just remove it (and the line above).